KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/CuePointsManager.java

```diff
@@ -276,139 +273,151 @@ public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketize
 				case APPLICATION_INSTANCE_STARTED:
 					KalturaApplicationInstanceEvent applicationInstanceEvent = (KalturaApplicationInstanceEvent) event;
 					onAppStart(applicationInstanceEvent.getApplicationInstance());
 					break;
-
 				default:
 					break;
 			}
 		}
 
-		if(event.getType() instanceof KalturaEventType){
+		if (event.getType() instanceof KalturaEventType) {
 
-			switch((KalturaEventType) event.getType())
-			{
+			KalturaMediaStreamEvent streamEvent = (KalturaMediaStreamEvent) event;
+			switch ((KalturaEventType) event.getType()) {
 				case STREAM_PUBLISHED:
-					KalturaMediaStreamEvent streamEvent = (KalturaMediaStreamEvent) event;
 					onPublish(streamEvent.getMediaStream(), streamEvent.getEntryId());
 					break;
-
+				case STREAM_UNPUBLISHED:
+					onUnPublish(streamEvent.getMediaStream(), streamEvent.getEntryId());
+					break;
 				default:
 					break;
 			}
 		}
+	}
+
+	private void onUnPublish(IMediaStream mediaStream, String entryId) {
+		final String streamName = mediaStream.getName();
+		Timer t;
+		synchronized (streams) {
+			if (!streams.containsKey(streamName)) {
+				logger.error("Unpublishing a stream that does not exist in streams map: " + streamName);
+				return;
+			}
+			t = streams.remove(streamName);
+		}
+		logger.debug("Stopping CuePoints timer for stream " + streamName);
+		t.cancel();
+		t.purge();
+
+		liveManager.removeReferrer(entryId, this);
 
-		super.onEvent(event);
 	}
 
 	protected void onAppStart(IApplicationInstance applicationInstance) {
-		logger.debug("Application instance started, adding live stream packetizer listener");
 		applicationInstance.addLiveStreamPacketizerListener(liveStreamPacketizerListener);
+		syncPointsInterval = applicationInstance.getProperties().getPropertyInt(KALTURA_SYNC_POINTS_INTERVAL_PROPERTY, DEFAULT_SYNC_POINTS_INTERVAL_IN_MS);
+		logger.info("CuePointsManager application instance started. Sync points interval: " + syncPointsInterval);
 	}
 
-	protected void onPublish(IMediaStream stream, String entryId) {
+	protected void onPublish(final IMediaStream stream, final String entryId) {
 		if(stream.getClientId() < 0){
 			logger.debug("Stream [" + stream.getName() + "] entry [" + entryId + "] is a transcoded rendition");
 			return;
 		}
 
-		logger.debug("Stream [" + stream.getName() + "] entry [" + entryId + "]");
-		ILiveStreamManager liveManager = KalturaServer.getManager(ILiveStreamManager.class);
+		String streamName = stream.getName();
+		logger.debug("Stream [" + streamName + "] entry [" + entryId + "]");
 		liveManager.addReferrer(entryId, this);
 
+		final Timer t;
 		synchronized (streams) {
-			ArrayList<IMediaStream> entryStreams;
-			if (streams.containsKey(entryId))
-				entryStreams = streams.get(entryId);
-			else
-				entryStreams =  new ArrayList<IMediaStream>();
-
-			entryStreams.add(stream);
-			streams.put(entryId, entryStreams);
+			if (streams.containsKey(streamName)) {
+				//TODO, can this situation occur? need to reset timer?
```

Comment:
this would mean that publish was called twice without unpublishing first - this needs to be handled upon stream creation in LiveStreamEntry before the publish event is raised.
Until it does, I suggest you replace .debug with .error